src/components/FeaturePanel/Climbing/GradeSystemSelect.tsx

```diff
@@ -20,17 +20,24 @@ import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
 import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
 import { t } from '../../../services/intl';
 import { ClimbingGradesTable } from './ClimbingGradesTable/ClimbingGradesTable';
+import { useVisibleGradeSystems } from './utils/useVisibleGradeSystems';
 
 type Props = {
   selectedGradeSystem: GradeSystem;
   setGradeSystem: (GradeSystem: GradeSystem) => void;
   allowUnsetValue?: boolean;
 };
 
-const GradeSystemItem = ({ isMinor, onClick, selectedGradeSystem }) => (
-  <>
-    {GRADE_SYSTEMS.filter(({ minor }) => minor === isMinor).map(
-      ({ key, name, description, flags }) => (
+const GradeSystemItem = ({ showMinor, onClick, selectedGradeSystem }) => {
+  const visibleGradeSystems = useVisibleGradeSystems();
+
+  return (
+    <>
+      {GRADE_SYSTEMS.filter(({ key }) =>
+        showMinor
+          ? !visibleGradeSystems.includes(key)
+          : visibleGradeSystems.includes(key),
+      ).map(({ key, name, description, flags }) => (
```

Comment:
Could you please extract this to a variable? It is helpful to not use any complicated logic in the render itself - a.k.a. separation of concerns.